### DISCUSSION TITLE
ZWN-RSM2 dual relay not updating device2 status via switch

### DISCUSSION THREAD

**Discussion by floatiepen** [zwave-js/zwave-js-ui#985] [Q&A]
I bought a ENERWAVE Z-Wave Plus Dual Relay Module (Device ID
282-1542-273 [(0x011a-0x0111-0x0606)](https://devices.zwave-js.io/?jumpTo=0x011a:0x0111:0x0606:0.0); ZWN-RSM2-PLUS - from [Amazon](https://www.amazon.com/gp/product/B07KQMGH7X)) and set it up, and at first, it seemed everything was great.  Each switch controls its device and they both popped up in zwavejs2mqtt and in turn in Home Assistant.  I can toggle the devices on/off via Home Assistant and all seemed great.

The trouble is that when Device2 is turned on at the switch, the z-wave object doesn't update.  Moving the Device1 switch does correctly update the z-wave status.  Is this a fault with the module? Or is there something in zwavejs2mqtt that isn't handling the update correctly?

In my setup, the module is node 7. When I look at Control Panel -> node 7 -> Binary Switch, I see the two booleans for currentValue:
[7-37-1-currentValue] -> updates when changed via z-wave or the physical switch (the expected behavior)
[7-37-2-currentValue] -> only updates when I toggle the device via z-wave (does not update when switch is moved)

I'm very new to z-wave and zwavejs2mqtt, so I'm not sure what more info to look at/provide to troubleshoot.  If I knew where to look for the device settings, I'd have a go to see if I could fix it.  maybe it's just a configuration quirk that needs handling.

Any tips on where to start to troubleshoot this?

**Comment by robertsLando** [comment]
@AlCalzone

**Comment by AlCalzone** [comment]
Please make a zwave-js log, loglevel `debug` or `silly` and attach it here as a file.

**Reply by AlCalzone** [reply]
This might be a misconfiguration, this might be an incorrect association, this might be a device bug, or this might be because HA uses an older version of zwave-js.

**Reply by floatiepen** [reply]
Attached is a silly-level log (I manually toggled each of the two switches on and off, but I don't see that here).

While I was poking around in the zwave-js devices page, I see the page linked for my device says it's for [ZWN-RSM1/ZWN-RSM2](https://devices.zwave-js.io/?jumpTo=0x011a:0x0111:0x0606:0.0), and there is a separate page for [ZWN-RSN1-PLUS](https://devices.zwave-js.io/?jumpTo=0x011a:0x0101:0x5605:0.0).  The relay I'm using is actually labeled ZWN-RSN2-PLUS.  Maybe it should actually be affiliated with the other ZWN-RSN1-PLUS (and there seems to be an entry missing that is present on the regular 1/2 page: Association Groups ID=3, I'm guessing that just like the other set, the -1 relay would just not use the extra association group)

Is there a way I can manually change this in my local setup to test it?

[zwavejs2mqtt - log.log](https://github.com/zwave-js/zwavejs2mqtt/files/6254979/zwavejs2mqtt.-.log.log)

**Reply by AlCalzone** [reply]
That is not a zwave-js log. => https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/bug_report?id=export-logs point 1

**Reply by floatiepen** [reply]
My mistake, I'm sorry about that.  thanks for your patience and assistance!  Also, I had tried to remove and re-add the module, so now it is node 11 in my system.  zwave-js log attached. (in the future, I just toggled each on/off a couple times should I filter it to try to capture only node 11?)

[zwavejs_1.log](https://github.com/zwave-js/zwavejs2mqtt/files/6258751/zwavejs_1.log)

**Reply by AlCalzone** [reply]
Looking at the manual:
> The device need creating an End Point association from lifeline Association Group to Root Device of gateway. 

this does not seem to be the case for your device. Can you please perform a re-interview of Node 11 and post another log of that?

**Reply by floatiepen** [reply]
Thanks for looking into this. Here is the log of the interview:
[zwavejs_1 - re-interview.log](https://github.com/zwave-js/zwavejs2mqtt/files/6273009/zwavejs_1.-.re-interview.log)

After that, I did another log of toggling the physical switch, and it only seems to mention endpoint 0.  Also, How long is the delay before the events are written to the log file on disk?  I think I cut some off (since I had toggled each switch twice, and I only see 4 events in the log).
[zwavejs_1 - 1 then 2.log](https://github.com/zwave-js/zwavejs2mqtt/files/6273016/zwavejs_1.-.1.then.2.log)

**Reply by AlCalzone** [reply]
Ok looks like this is because of the associations. Can you add Node ID 1 to both group 2 and 3 and check if that is better?

**Reply by floatiepen** [reply]
That did it!  Both lights now work and respond to the change at the switch.

**Reply by AlCalzone** [reply]
Thanks, I raised https://github.com/zwave-js/node-zwave-js/issues/2280

**Reply by floatiepen** [reply]
You have been very helpful, thanks again.  And than you for submitting that other issue, I hope it gets resolved and makes life easier for future users.

**Comment by ronytomen** [comment]
Submitted PR https://github.com/zwave-js/node-zwave-js/pull/4046
